### PR TITLE
Address review comments in old PRs

### PR DESCRIPTION
**Pull request by bhagyasakalanka**

### CODE REVIEW

**Review by somindatommy — APPROVED** [review]

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 1]
```diff
@@ -706,32 +706,33 @@ private String getCaptchaParams(String tenantDomain, int failedLoginAttempts) {
...
+                        Property[] maxFailedConfig =
+                                BasicAuthenticatorDataHolder.getInstance().getIdentityGovernanceService()
+                                        .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName},
+                                                tenantDomain);
+                        Property maxFailedProperty = maxFailedConfig[0];
+                        try {
+                            if (Integer.valueOf(maxFailedProperty.getValue()) >= failedLoginAttempts) {
+                                if (log.isDebugEnabled()) {
+                                    log.debug("Number of failed attempts is less than max failed " +
+                                            "attempts for reCaptcha. Recaptcha will not be enforced");
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:724]: Condition says `Integer.valueOf(maxFailedProperty.getValue()) >= failedLoginAttempts`

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 2]
```diff
@@ -706,32 +706,33 @@ private String getCaptchaParams(String tenantDomain, int failedLoginAttempts) {
...
-                            } catch (NumberFormatException e) {
-                                log.error(String.format("Invalid Max failed attempts for reCaptcha: %s",
-                                        maxFailedProperty.getValue()), e);
+                                continue;
                             }
-                        } else {
                             if (log.isDebugEnabled()) {
-                                log.debug("Enforcing recaptcha for SSO Login is not enabled.");
+                                log.debug("Number of failed attempts is higher than max failed " +
+                                        "attempts for reCaptcha. Recaptcha will be enforced");
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:730]: The log is wrong right? It should be lesser than max attempts right?

**Review by somindatommy — CHANGES_REQUESTED** [review]

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java (hunk 1 above)
Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:724]: Condition is correct right? It is the negation of the previous condition `Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts`.  We check if the max failed attempts is higher or equals to failed login attempts and indicate recaptcha will not be enforced.

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java (hunk 2 above)
Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:730]: No this is the true condition when failed login attempts exceeds the max failed attempt.  So log says, reCaptcha will be enforced.

**Review by somindatommy — APPROVED** [review]

**Pull request merged**